sky/packages/sky/lib/gestures/scale.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:sky' as sky;
+
+import 'package:sky/gestures/arena.dart';
+import 'package:sky/gestures/recognizer.dart';
+import 'package:sky/gestures/constants.dart';
+
+enum ScaleState {
+  ready,
+  possible,
+  accepted,
+  started
+}
+
+typedef void GestureScaleStartCallback(sky.Point focalPoint);
+typedef void GestureScaleUpdateCallback(double scale, sky.Point focalPoint);
```

Comment:
Do you think its better to give the two focal points or for each update to give you the delta to the focal point since the last update?  Drag gives you the delta...